src/components/tree/infoPanels/click.js

```diff
@@ -248,6 +248,18 @@ const Trait = ({node, trait, colorings, isTerminal}) => {
  * @param  {function} props.t
  */
 const NodeClickedPanel = ({selectedNode, nodesLhsTree, nodesRhsTree, clearSelectedNode, colorings, observedMutations, geneSortFn, tipLabelKey, t}) => {
+  useEffect(() => {
+    function handleEscapeKey(event) {
+      if (event.key==="Escape" && selectedNode) {
+        clearSelectedNode(selectedNode);
+      }
+    }
+    document.addEventListener('keyup', handleEscapeKey);
+    return function cleanup() {
+      document.removeEventListener('keyup', handleEscapeKey);
+    };
+  }, [selectedNode, clearSelectedNode]);
+
```

Comment:
One salient difference is that this will add/remove listeners every time a modal appears/disappears as opposed to a single listener set up when the tree panel mounts. In reality this won't matter because opening a modal is a one-off action, but it's something to be aware of when designing listeners.

An improvement would be to return early to avoid adding an unnecessary listener when we _close_ the modal:
```js
  useEffect(() => {
    if (!selectedNode) return
    ...
```

Reply:
Thanks for noting the difference. There is a way to retain the existing behavior using a ref, which I've incorporated into the main commit (c136e2588239fe54e6e82e5043c584b852ba9480). I added another commit to switch to the plain `useEffect` pattern with your proposed improvement above (eae3fbad08f44e60f85639fa94f81a1a6ef73a45), which we can drop if you think the existing behavior is better.

Reply:
I need to dive back into the world of hooks one day, although I wasn't very enamored with them. Perhaps [the rest of the group feel differently!](https://github.com/nextstrain/auspice/pull/2018#discussion_r2474387773) Is adding an additional `useRef` & `useEffect` the recommended way? I find it somewhat convoluted to read, although I can figure out how it works. Up to you whether this is your preferred method vs. a single but slightly more costly `useEffect` - either works for me.